hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -423,6 +423,14 @@ private void saveWorkloadProfileMetadataToInflight(WorkloadProfile profile, Hood
       logger.info("RDD PreppedRecords was persisted at: " + preppedRecords.getStorageLevel());
     }
 
+    // Cache Latest File-System View & Seal
+    preppedRecords.map(rec -> rec.getPartitionPath()).distinct().collect().stream().forEach(partitionPath -> {
+      // Populate latest version file-slice cache. No-op if already populated
+      long numFileSlices = hoodieTable.getLatestFileSliceOnlyFSView().getLatestFileSlices(partitionPath).count();
```

Comment:
How does the populated values on executor make its way back to driver?

Reply:
+1

Reply:
ohh I see, the collect() is before you perform the getLatestFileSlice map operations.

Reply:
I am moving this part to a separate utility class to avoid confusion and be reusable across places.